builtin/providers/ultradns/resource_ultradns_dirpool.go

```diff
@@ -0,0 +1,612 @@
+package ultradns
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/Ensighten/udnssdk"
+	"github.com/fatih/structs"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/mitchellh/mapstructure"
+)
+
+func resourceUltradnsDirpool() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceUltradnsDirpoolCreate,
+		Read:   resourceUltradnsDirpoolRead,
+		Update: resourceUltradnsDirpoolUpdate,
+		Delete: resourceUltradnsDirpoolDelete,
+
+		Schema: map[string]*schema.Schema{
+			// Required
+			"zone": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				// 0-255 char
+			},
+			"rdata": &schema.Schema{
+				Type:     schema.TypeSet,
```

Comment:
Does this have to be a set or can it be a list? Is there a max size of the items that can be supplied?